platforms/shared/shared.cmake

```diff
@@ -1,26 +1,82 @@
 # Copyright (C) 2017 Prism Skylabs
-macro(findBoostCurl)
-    set(Boost_USE_STATIC_LIBS ON)
-    set(Boost_USE_MULTITHREADED ON)
-    set(Boost_USE_STATIC_RUNTIME ON)
+function(download_artifact repoUrl platform projectName version dstPath)
+
+    set(archiveName artifact.tar.gz)
 
-    find_package(Boost 1.60.0 REQUIRED COMPONENTS chrono)
+    #check if artifact already exists locally
+    if(EXISTS ${dstPath}/${archiveName})
+       message(STATUS "Artifact exists, skipping download")
+       return()
+    endif()
+
+    message(STATUS "Making download URL for artifact ${repoUrl}, ${projectName}, ${platform}, ${version}")
+    set(downloadUrl "${repoUrl}/${projectName}/${platform}/${projectName}_${version}_${platform}.Release.tar.gz")
+    message(STATUS "Download URL: ${downloadUrl}")
 
-    if (NOT Boost_FOUND)
-        if (NOT DEFINED ENV{BOOST_ROOT})
-            message (FATAL_ERROR "Please, define BOOST_ROOT environment variable")
-        endif ()
+    file(MAKE_DIRECTORY ${dstPath})
 
-        set (BOOST_ROOT $ENV{BOOST_ROOT})
+    file(DOWNLOAD ${downloadUrl} ${dstPath}/${archiveName}
+        STATUS status
+        SHOW_PROGRESS)
 
-        find_package(Boost 1.60.0 REQUIRED COMPONENTS chrono)
+    list (GET status 0 retCode)
+    list (GET status 1 retMessage)
+
+    if (NOT "${retCode}" STREQUAL "0")
+        message(FATAL_ERROR "Error downloading artifact: ${retCode} ${retMessage}")
+        execute_process(COMMAND rm -rf ./* WORKING_DIRECTORY ${dstPath})
     endif()
 
+    execute_process(
+       COMMAND tar -xvzf ${archiveName}
```

Comment:
Have we considered using cmake's builtins https://cmake.org/cmake/help/v3.2/manual/cmake.1.html#command-line-tool-mode ?